demo/brush-with-arguments.html

```diff
@@ -0,0 +1,78 @@
+<!doctype html>
+<head>
+    <meta content="text/html;charset=utf-8" http-equiv="Content-Type">
+    <meta content="utf-8" http-equiv="encoding">
+</head>
+<title>Brushing Example</title>
+<link rel="stylesheet" type="text/css" href="./parcoords.css">
+<link rel="stylesheet" type="text/css" href="style.css">
+<script src="./lib/d3.v5.min.js"></script>
+<script src="./parcoords.standalone.js"></script>
+<div id="example" class="parcoords" style="width:960px;height:200px;"></div>
+<p>Loads an external <a href="data/cars.csv">csv file</a>, creates a custom <a
+        href="https://github.com/mbostock/d3/wiki/Quantitative-Scales#wiki-quantitative">quantitative color scale</a>
+    using <a href="http://bl.ocks.org/3014589">L*a*b interpolation</a>, and enables brushing.
+
+<h3>Brush Debug</h3>
+<p id="brush-results" style="background-color:#ccc;">
+</p>
+
+
+    <script>
+        var parcoords = ParCoords()("#example")
+        //    .color(color)
+            .alpha(0.4)
+            .on("brushstart", function(brushed, args){
+              if(args !== undefined) {
+                d3.select("#brush-results").html([
+                  '<span style="font-weight:bold;">brush end</span>',
+                  'dimension: ' + args[0]
+                ].join("<br/>"))
+              } else {
+                d3.select("#brush-results").html('cleared brush')
+              }
+            })
+            .on("brush", function(brushed, args){
+              if(args !== undefined) {
+                var brush_selection = d3.brushSelection(args[2][0])
```

Comment:
It is good to add more information to brush callback.

In your demo, you use the brush selection in this way:
```
// 1) get brushSelection from args
var brush_selection = d3.brushSelection(args[2][0])
// 2) map selection to dimension values
brush_selection.map(d=>this.dimensions()[args[0]].yscale.invert(d))
```
Would it be better to return dimension values directly from args so that end users don't need to access yScale?
```
var brushRanges = d3.brushSelection(args[2][0])
// brushRanges is  [6.658536585365853,5.621951219512195]
```
But that is based on a assumption that end users don't use `brushSelection` for other purposes.

Or, we can do: 
```
var brushRanges = d3.brushSelection(args[2][0])
// brushRanges is {
raw: [45,79]
data:  [6.658536585365853,5.621951219512195]
}
```

What do you think?

Reply:
@BigFatDog I like the idea a lot and makes sense especially since this version is modular and many users will not also have the full `d3`.  The yscale-transformed brush selection is definitely the most important, but I could envision a special case where the raw coordinates are useful.

I think providing the following data `on('brush'` would satisfy most users' demands:

1. dimension name or the dimension object
2. brush g svg node
3. brushSelection coordinates
4. brushSelection y-scale transformed coordinates

Is providing each of these as additional arguments ok, or do you think this should all be wrapped up in an object and provided as one argument?

Reply:
The data that you've described is good enough for this API. I agree with you on it.

Both of the two api styles are fine to me. I tend to choose approach 1 because it has more explicit arguments. As you have illustrated, the api style look like:

```
pc.on('brush', brushed, dimension, g, coordinates)
```
coordinates are:

```
I don't have any idea on naming..
[{
  raw: [2, 4] //raw coordinates, 
  data: [33, 252] //y-scale transformed
}]
```
brushSelection coordinates and transformed coordinates in yScale are merged because they have `one-to-one` relationship. 
Or we can provide the two coordinates separately:
```
pc.on('brush', brushed, dimension, g, rawCoordinates, transformedCoordinates)
```
Which one do you prefer?

Reply:
Not a strong preference, but since raw and transformed are 1:1, I think I like the object approach better.  Enjoyed working with you on this.

Reply:
I have nothing else.  
That's my pleasure. 😃

Reply:
@BigFatDog anything you would like me to do to complete or close this pull?

Reply:
I've been waiting changes to API based on our discussion. My plan is to merge this pull request after api is revised

Reply:
Did I misunderstand anything? I wanted to ping you regarding this pull request for one or two times but I hesitated, because I don't want to make you feel pressure from open-source project (and shouldn't be). Meanwhile, I think it is quite a natural thing that you are busy with your work and get back to this pull request when you have time.